### ISSUE TITLE
CentOS7 - Missing build dependency cracklib-devel

### ISSUE THREAD

**Issue by arekkusu** [ltb-project/openldap-rpm#15]
Got following error when build the RPM on Centos7 without cracklib-devel installed. Works fine after installing the package.

```
+ cd ltb-project-openldap-ppolicy-check-password-1.1
+ make CONFIG=/usr/local/openldap/etc/openldap/check_password.conf 'LDAP_INC=-I../include -I../servers/slapd'
rm -f check_password.o check_password.so check_password.lo
rm -f -r .libs
gcc -g -O2 -Wall -fpic -DHAVE_CRACKLIB -DCRACKLIB_DICTPATH="\"/usr/share/cracklib/pw_dict\"" -DCONFIG_FILE="\"/usr/local/openldap/etc/openldap/check_password.conf\"" -DDEBUG -c -I../include -I../servers/slapd  check_password.c
check_password.c:13:19: fatal error: crack.h: No such file or directory
 #include "crack.h"
                   ^
compilation terminated.
make: *** [check_password.o] Error 1
error: Bad exit status from /var/tmp/rpm-tmp.4PYD4v (%build)
```

Edited: I noticed 'cracklib-devel' is actually mentioned here. But not working as intended...

```
#=================================================
# Subpackage check-password
#=================================================
%package check-password
Summary:        check_password module for password policy
Version:        %{check_password_version}
Release:        8%{?dist}
Group:          Applications/System
URL:            http://www.ltb-project.org

%if "%{?dist}" == ".el6"
BuildRequires:  cracklib-devel
%endif
%if "%{?dist}" == ".el7"
BuildRequires:  cracklib-devel
%endif

Requires:       cracklib, cracklib-dicts, %{real_name}-ltb >= %{real_version}
```

**Comment by coudot** [comment]
Thanks for the report

**Comment by arekkusu** [comment]
In centos7 the dist value is `.el7.centos` instead of `.el7`, hence the macro is failing.

No idea why they did that, are people also run into the issue:

https://cstan.io/?p=8946&lang=en
https://serverfault.com/questions/688485/rpm-dist-tag-not-behaving-as-documented

Following proposed fix seems to work. Again only tested on Centos7.4

```
%if 0%{?rhel} == 7
 %define dist .el7
%endif
```

**Comment by coudot** [comment]
You can also define this in your own rpmmacros file

**Comment by coudot** [comment]
Fixed by #19
